library/common/jni/jni_utility.cc

```diff
@@ -113,11 +113,28 @@ jobject native_map_to_map(JNIEnv* env, envoy_map map) {
 }
 
 envoy_data buffer_to_native_data(JNIEnv* env, jobject j_data) {
-  size_t data_length = static_cast<size_t>(env->GetDirectBufferCapacity(j_data));
-  return buffer_to_native_data(env, j_data, data_length);
+  // Returns -1 if the buffer is not a direct buffer.
+  jlong data_length = env->GetDirectBufferCapacity(j_data);
+
+  if (data_length < 0) {
```

Comment:
I really missed a step with my "ultra safe refactoring"...

Thanks!

Reply:
No worries - I actually think our integration tests would have caught it; they're just unfortunately turned off at the moment. I intend to fix them up and re-enable them in the near future. :)

Reply:
The integration tests do currently run on linux: https://github.com/envoyproxy/envoy-mobile/blob/main/.github/workflows/android_tests.yml#L134